File: LeetCode/Hao/1TwoSum.java
public class Solution {
    public int[] twoSum(int[] nums, int target) {
        int[] result = new int[]{-1, -1};
        Map<Integer, Integer> m = new HashMap<Integer, Integer> ();
        
        for (int i = 0; i < nums.length; i ++) {
            int k = target - nums[i];
            
            if (m.containsKey(k) == true) {
                result[0] = m.get(k);
                result[1] = i + 1;
            } else {
                m.put(nums[i], i + 1);
            }
            
        }
        
        return result;

    }
}

File: LeetCode/Peng/1TwoSum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        int n = nums.size();
        vector<int> result;
        unordered_map<int, int> numCounter;
        for (int i = 0; i < n; i++) {
            int current = target - nums.at(i);
            if (numCounter.count(current)) {
                result.push_back(i + 1);
                result.push_back(numCounter[current] + 1);
                sort(result.begin(), result.end());
                break;
            }
            numCounter[nums.at(i)] = i;
        }
        return result;
    }
};